app/src/test/java/io/apicurio/registry/RegistryClientTest.java

```diff
@@ -188,6 +188,61 @@ public void testLabels(Supplier<RegistryService> supplier) throws Exception {
         }
     }
 
+    @RegistryServiceTest
+    void nameOrderingTest(Supplier<RegistryService> supplier) throws Exception {
+
+        final String firstArtifactId = generateArtifactId();
+        final String secondArtifactId = generateArtifactId();
+        RegistryService client = supplier.get();
+
+        try {
+
+            // warm-up
+            client.listArtifacts();
+
+            String name = "aaaTestorder" + ThreadLocalRandom.current().nextInt(1000000);
+            ByteArrayInputStream artifactData = new ByteArrayInputStream(
+                    ("{\"type\":\"record\",\"title\":\"" + name + "\",\"fields\":[{\"name\":\"foo\",\"type\":\"string\"}]}")
+                            .getBytes(StandardCharsets.UTF_8));
+
+            CompletionStage<ArtifactMetaData> cs = client.createArtifact(ArtifactType.JSON, firstArtifactId, null, artifactData);
+            long id = ConcurrentUtil.result(cs).getGlobalId();
+
+            retry(() -> {
+                ArtifactMetaData artifactMetaData = client.getArtifactMetaDataByGlobalId(id);
+                Assertions.assertNotNull(artifactMetaData);
+            });
+
+            String secondName = "bbbTestorder" + ThreadLocalRandom.current().nextInt(1000000);
+            ByteArrayInputStream secondData = new ByteArrayInputStream(
+                    ("{\"type\":\"record\",\"title\":\"" + secondName + "\",\"fields\":[{\"name\":\"foo\",\"type\":\"string\"}]}")
+                            .getBytes(StandardCharsets.UTF_8));
+
+            CompletionStage<ArtifactMetaData> secondCs = client.createArtifact(ArtifactType.JSON, secondArtifactId, null, secondData);
+            long secondId = ConcurrentUtil.result(secondCs).getGlobalId();
 
+            retry(() -> {
+                ArtifactMetaData artifactMetaData = client.getArtifactMetaDataByGlobalId(secondId);
+                Assertions.assertNotNull(artifactMetaData);
+            });
```

Comment:
Also maybe throw in an artifact with no name.  In that case the *ID* should be used as the name.